core/src/main/java/org/elasticsearch/index/mapper/BaseGeoPointFieldMapper.java

```diff
@@ -183,16 +192,19 @@ public Y build(Mapper.BuilderContext context) {
             context.path().remove();
 
             return build(context, name, fieldType, defaultFieldType, context.indexSettings(),
-                    latMapper, lonMapper, geoHashMapper, multiFieldsBuilder.build(this, context), ignoreMalformed(context), copyTo);
+                latMapper, lonMapper, geoHashMapper, multiFieldsBuilder.build(this, context), ignoreMalformed(context), copyTo);
         }
     }
 
     public abstract static class TypeParser implements Mapper.TypeParser {
         @Override
         public Mapper.Builder<?, ?> parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
             Builder builder;
-            if (parserContext.indexVersionCreated().before(Version.V_2_2_0)) {
+            Version indexVersionCreated = parserContext.indexVersionCreated();
+            if (indexVersionCreated.before(Version.V_2_2_0)) {
                 builder = new LegacyGeoPointFieldMapper.Builder(name);
+            } else if (indexVersionCreated.onOrAfter(Version.V_5_0_0_alpha6)) {
```

Comment:
looks like 5.0 beta 1 isn't in master? Can I take care of this when cherry-picking to the 5.0/5.x branches or should we cherry-pick 3f6273f92ee5d18b78cfbfafe7c38dff46f29213 to master?